mpegts-segmenter/src/segmenter.rs

```diff
@@ -163,6 +167,7 @@ impl<S: SegmentStorage> Segmenter<S> {
                     pcr: self.pcr.unwrap_or(0),
                     pts: None,
                     bytes_written: 0,
+                    temi_timeline_descriptor,
```

Comment:
I think you could do something like this to avoid unnecessary clones:

```suggestion
                    temi_timeline_descriptor: af.and_then(|af| af.temi_timeline_descriptors.first()),
```

Reply:
`first()` returns and a reference, we need ownership.